Marlin/src/MarlinCore.cpp

```diff
@@ -1647,6 +1647,31 @@ void setup() {
 
   marlin_state = MF_RUNNING;
 
+  #if ENABLED(MARLIN_DEV_MODE)
+    SString<20> str20;
+    str20 = F("F-string, ");
+    str20.append(1234.5f).append(',').append(' ')
+         .append(2345.67).append(',').append(' ')
+         .echoln();
+
+    str20.concat(8).echoln();
+
+    TSS(repchr_t('-', 100)).echoln();
+
+    SString<100> str(F("Hello"));
```

Comment:
Depending on how many places the SIZE template parameter are using inside the string class, would not the compiler generate a lot of "duplicated" code, increasing a lot the size of the binary?
We know that each template instance is a new type with its code, but I don't know how exactly how compilers optimize it to avoid a lot of duplicate code.
Did you checked the generated code for some instances of this String template?

Depending of the impact, a some tweaks or even a dynamic buffer would be needed.

Reply:
The compiler resolves everything in-place, so there are no classes being statically generated anywhere, just the resulting code that builds and works with the string wherever it is used. The one thing that makes the code a little larger is that the string always uses `strnfunc` calls instead of `strfunc` calls to prevent buffer overruns in the case of an unterminated or over-long source string. That could be switched with a flag, making the code smaller but potentially less stable.